### DISCUSSION TITLE
Provide a connection string out of the box

### DISCUSSION THREAD

**Discussion by bennettdams** [planetscale/discussion#15] [Feature Requests] [answered]
I really like the idea behind Planet Scale. Using a serverless out of the box SQL database with a free tier to start a project with is exactly what I was looking for.

There's one basic thing that confused me when I signed up today:
After creating a database via the web UI, there was no connection string. I understand that Planet Scale is way more than a simple database host, but for a lot of projects a simple connection string for a default database (`main`) is more than enough to start. No need to download the CLI or configure something, just connect.

Is this something you're planning to do?

**Comment by cboothe** [comment]
I also need this. I have serverless functions that I want to connect to the "serverless db".
They can't require a sidecar pattern or proxy to connect to the database.

**Reply by mscoutermarsh** [reply]
Hi @cboothe. How are you running your functions? Lambda?

**Reply by cboothe** [reply]
@mscoutermarsh I wanted to try this from a Cloidflare worker

**Reply by tbarn** [reply]
Hey @cboothe! It's out now: https://planetscale.com/blog/connect-any-mysql-client-to-planetscale-using-connection-strings. I'd love to hear how it works with Cloudflare workers.

**Comment by Anders-PlanetScale** [comment]
For the time being you will need to use pscale CLI to access your databases. You can use either a localhost proxy created using the `pscale connect` command or use `pscale shell` command to get a mysql command prompt.

Our engineers are looking into options to add a direct connection string. It is proving to be a common feature request, and we will follow up as we have more information.

**Reply by shaunchurch** [reply]
There is some obsolete documentation[1] still on the website for an old version that seems to have supported a connection string / host, but that no longer appears to be supported (maybe why it's a popular request?). Disappointed to find this out after spending a bunch of time playing with it. I run most of my stuff in serverless environments, so requiring the CLI is a total showstopper. 

Look forward to seeing the return of the connection string in future!

[1] https://www.planetscale.com/blog/how-to-set-up-a-stateful-web-app-in-10-minutes

**Reply by mscoutermarsh** [reply]
Thanks Shaun. Will take care of that.

What platform are you trying to deploy to?

**Reply by shaunchurch** [reply]
Mainly Vercel. Hopefully you're going to tell me there's a way? :)

**Reply by mscoutermarsh** [reply]
Working on it! Node.js I'm guessing?

**Reply by arist0tl3** [reply]
I'm doing the same -- looking to deploy an express backend to vercel with a pscale connection. Happy to help test/answer anything about my env or setup!

**Reply by shaunchurch** [reply]
> Working on it! Node.js I'm guessing?

Node.js, Next.js, Blitz.js! :)

**Comment by chifi49** [comment]
Same interest here as well. I ve been using cloudflare workers and aws lambda lately and we need a direct connection string.

**Reply by mscoutermarsh** [reply]
Thanks @chifi49. Yes, we definitely want to support workers/lambda. We're working on making this easy for people.

**Reply by tbarn** [reply]
Hey @chifi49! It's out now: https://planetscale.com/blog/connect-any-mysql-client-to-planetscale-using-connection-strings

**Comment by victorybiz** [comment]
I also need this to connect a Laravel app to PlanetScale.

**Reply by tbarn** [reply]
Hey @victorybiz! This feature is out now: https://planetscale.com/blog/connect-any-mysql-client-to-planetscale-using-connection-strings. I'd love to get your feedback on how it works with one of your Laravel apps. Let me know if you need any help!

**Comment by irvinebroque** [comment]
+1 to this thread, even more helpful would be to provide an https interface that eliminated the need for the mysql javascript client entirely.

1. This would make it possible to use Planetscale with Cloudflare Workers without waiting for Workers to add node.js support. 
2. Abstracting away connection pooling entirely from client-land would make Planetscale competitive with RDS Data Service + Aurora Serverless (https://aurora-pg-lab.workshop.aws/lab10-aurora-serverless/lambda.html)

**Reply by danaugrs** [reply]
I strongly agree.

**Reply by shreyfirst** [reply]
agree

**Comment by tessro** [comment]
We ran into this today, as we are trying to use Planetscale from Netlify. Unfortunately it seems to be impossible, which has us looking for other solutions. :(

**Reply by irvinebroque** [reply]
Proxy function with Lamba + API Gateway worked as a stopgap measure for me, YMMV

**Reply by Phanatic** [reply]
Hey @paulrosania,

Thanks for your patience. We're getting very close to finishing the Connections Strings feature. 
I know it's painful right now, please hang tight and you should be able to connect to planetscale without the pscale CLI soon. 

Thanks,
Phani

**Reply by tessro** [reply]
Thanks! We were able to get up and running on Vercel which looks like it'll keep us using PlanetScale. 😅

**Reply by muhaimincs** [reply]
@Phanatic looking forward for the release. I'm hanging tight here

**Reply by heymartinadams** [reply]
@Phanatic was wondering if you might have some insight into this https://github.com/prisma/prisma/issues/7292#issuecomment-890269439

**Reply by abriginets** [reply]
@Phanatic do you have any specific dates? It is going to be released in August?

**Reply by dangdennis** [reply]
We're all sitting on the edges of our seat for this.

**Reply by abriginets** [reply]
I would even agree to get billed fullprice to get this 😄

**Reply by shreyfirst** [reply]
well well well don't give them ideas!

**Reply by Phanatic** [reply]
@abriginets Yes, it's coming in August.

**Reply by heymartinadams** [reply]
@Phanatic do you have an approx. ETA (and the likelihood of meeting that ETA)? We’re considering holding off on our beta launch so we can use PlanetScale from the get-go, rather than migrate down the road (from AWS, thank you!).

**Reply by Phanatic** [reply]
This feature is out now https://planetscale.com/blog/connect-any-mysql-client-to-planetscale-using-connection-strings

**Comment by shreyfirst** [comment]
Howd you do it?

**Reply by tessro** [reply]
Just using the integration documented here: https://vercel.com/integrations/planetscale

**Reply by magic-akari** [reply]
> Just using the integration documented here: https://vercel.com/integrations/planetscale

Hello @paulrosania 
There seems to be an error in the demo.

https://planetscale-next-js.preview.planetscale.com
<img width="917" alt="image" src="https://user-images.githubusercontent.com/7829098/128461348-c4eaf495-2f90-4c66-aab9-f61f62aeb2d0.png">

**Comment by Phanatic** [comment, marked as answer]
We have now released this feature for everyone to use. https://planetscale.com/blog/connect-any-mysql-client-to-planetscale-using-connection-strings

**Reply by abriginets** [reply]
You are awesome! I sure will tell my friends and colleagues about you. Just made my development process a ton easier...

**Comment by irvinebroque** [comment]
Docs for working with AWS Lambda + mysql2 would be really helpful here. The current docs don't mention Lambda at all, but it's one of the trickiest to get certificates working on, since to do it right you have to use AWS Secrets Manager and jump through a few hoops to connect everything. Would save a lot of time to have a simple working example.

**Reply by tbarn** [reply]
thanks @irvinebroque for calling that out! what's your AWS Lambda setup like right now?

**Reply by irvinebroque** [reply]
Currently using Lambda exclusively as a proxy to connect Cloudflare Workers to Planetscale, because Workers don't have full node.js runtime support yet and aren't able to connect directly to Plantescale. It's a pain, and slower than it'd ideally be, but workable for now.

I think the ideal example would focus on how to deal with Certificate Authority using Lambda, since Amazon's own docs for this are such a hassle to navigate and find, and it involves permissions, roles, etc.

In the long-run I really think it'd be advantageous to Planetscale to provide a pure https interface to Plantescale, similar to what a lot of cloud NoSQL companies provide. AFAIK that'd make Planetscale the *only* SQL-based DBaaS to offer this in a way that isn't tied up to a specific vendor - AWS does this with the RDS Data API but it only works within the AWS ecosystem. I really think you're so close to unlocking a ton of potential with Cloudflare Workers + Planetscale, but might be undervaluing how much energy there is right now behind Cloudflare Workers, and overvaluing Lambda a bit, for people starting new projects.

**Reply by cboothe** [reply]
Thanks @irvinebroque I have run into this exact same issue. I am already using CF workers to AWS SQS. But want to route to Planetscale directly too. Essentially replace FaunaDB with Planetscale.
